pkg/payload/task_test.go

```diff
@@ -0,0 +1,169 @@
+						"metadata": map[string]interface{}{
+							"name": "file-json",
+						},
+					},
+				},
+			}},
+			want: ` "file-json" (0 of 0)`,
+		},
+		{
+			name: "Manifest with GVK",
+			task: &Task{Manifest: &lib.Manifest{
+				GVK: schema.GroupVersionKind{Group: "extensions", Version: "v1beta1", Kind: "Ingress"},
+				Obj: &unstructured.Unstructured{
+					Object: map[string]interface{}{
+						"kind":       "Test",
+						"apiVersion": "v1",
+						"metadata": map[string]interface{}{
+							"name": "file-json",
+						},
+					},
+				},
+			}},
+			want: `ingress "file-json" (0 of 0)`,
+		},
+		{
+			name: "Manifest with name and namespace",
+			task: &Task{Manifest: &lib.Manifest{
+				OriginalFilename: "a",
+				Obj: &unstructured.Unstructured{
+					Object: map[string]interface{}{
+						"kind":       "Test",
+						"apiVersion": "v1",
+						"metadata": map[string]interface{}{
+							"name":      "file-json",
+							"namespace": "foo",
+						},
+					},
+				},
+			}},
+			want: ` "foo/file-json" (0 of 0)`,
+		},
+		{
+			name: "Manifest with GVK and namespace",
+			task: &Task{Manifest: &lib.Manifest{
+				GVK: schema.GroupVersionKind{Group: "extensions", Version: "v1beta1", Kind: "Ingress"},
+				Obj: &unstructured.Unstructured{
+					Object: map[string]interface{}{
+						"kind":       "Test",
+						"apiVersion": "v1",
+						"metadata": map[string]interface{}{
+							"name":      "file-json",
+							"namespace": "foo",
+						},
+					},
+				},
+			}},
+			want: `ingress "foo/file-json" (0 of 0)`,
+		},
+		{
+			name: "index and total",
+			task: &Task{
+				Manifest: &lib.Manifest{
+					GVK: schema.GroupVersionKind{Group: "extensions", Version: "v1beta1", Kind: "Ingress"},
+					Obj: &unstructured.Unstructured{
+						Object: map[string]interface{}{
+							"kind":       "Test",
+							"apiVersion": "v1",
+							"metadata": map[string]interface{}{
+								"name": "file-json",
+							},
+						},
+					},
+				},
+				Index: 3,
+				Total: 7,
+			},
+			want: `ingress "file-json" (3 of 7)`,
+		},
+		{
+			name: "Namespaced, index and total",
+			task: &Task{
+				Manifest: &lib.Manifest{
+					GVK: schema.GroupVersionKind{Group: "extensions", Version: "v1beta1", Kind: "Ingress"},
+					Obj: &unstructured.Unstructured{
+						Object: map[string]interface{}{
+							"kind":       "Test",
+							"apiVersion": "v1",
+							"metadata": map[string]interface{}{
+								"name":      "file-json",
+								"namespace": "foo",
+							},
+						},
+					},
+				},
+				Index: 5,
+				Total: 10,
+			},
+			want: `ingress "foo/file-json" (5 of 10)`,
+		},
+		{
+			name: "List",
+			task: &Task{Manifest: &lib.Manifest{
+				OriginalFilename: "list_manifest.yaml",
+				Obj: &unstructured.Unstructured{
+					Object: map[string]interface{}{
+						"kind":       "List",
+						"apiVersion": "v1",
+						"items": []interface{}{
+							"foo",
+							"bar",
+						},
+					},
+				},
+			}},
+			want: ` "list_manifest.yaml" (0 of 0)`,
```

Comment:
Leading space when we have no GKV isn't great, but that's not new in this PR, and this is much better than the [current][1]:

```
Could not update list \"\" (188 of 529)"
```

for lists.

/lgtm

[1]: https://bugzilla.redhat.com/show_bug.cgi?id=1790629#c0